recipes/zlib-bytesread-to-byteswritten/src/workflow.ts

```diff
@@ -0,0 +1,140 @@
+import { getNodeRequireCalls } from "@nodejs/codemod-utils/ast-grep/require-call";
+import { getNodeImportStatements } from "@nodejs/codemod-utils/ast-grep/import-statement";
+import { resolveBindingPath } from "@nodejs/codemod-utils/ast-grep/resolve-binding-path";
+import { removeLines } from "@nodejs/codemod-utils/ast-grep/remove-lines";
+import type { Edit, Range, SgRoot } from "@codemod.com/jssg-types/main";
+import type Js from "@codemod.com/jssg-types/langs/javascript";
+
+const ZLIB_FACTORIES = [
+	"createGzip",
+	"createGunzip",
+	"createDeflate",
+	"createInflate",
+	"createBrotliCompress",
+	"createBrotliDecompress",
+	"createUnzip",
+];
+
+export default function transform(root: SgRoot<Js>): string | null {
+	const rootNode = root.root();
+	const edits: Edit[] = [];
+	const linesToRemove: Range[] = [];
+
+	// 1 Find all static zlib imports/requires
+	const importNodes = [
+		...getNodeRequireCalls(root, "node:zlib"),
+		...getNodeImportStatements(root, "node:zlib")
+	];
+
+	const factoryBindings: string[] = [];
+	const streamVariables: string[] = [];
+
+	for (const node of importNodes) {
+		const baseBind = resolveBindingPath(node, "$");
+
+		if (baseBind) {
+			for (const factory of ZLIB_FACTORIES) factoryBindings.push(`${baseBind}.${factory}`);
+		}
+		for (const factory of ZLIB_FACTORIES) {
+			const binding = resolveBindingPath(node, `$.${factory}`);
+			if (binding && !factoryBindings.includes(binding)) factoryBindings.push(binding);
+		}
+	}
+
+	// 1.b Handle dynamic imports: `await import("node:zlib")`
+	const dynamicImports = rootNode.findAll({ rule: { pattern: "const $$$VAR = await import($$$MODULE)" } });
+	for (const imp of dynamicImports) {
+		const moduleName = imp.find({
+			rule: {
+				kind: "string_fragment",
+				inside: {
+					kind: "string",
+					inside: {
+						kind: "arguments",
+					}
+				}
+			}
+		})?.text();
+
+		if (moduleName === "node:zlib") {
+			const varName = imp.find({
+				rule: {
+					kind: "identifier",
+					inside: {
+						kind: "variable_declarator",
+					}
+				}
+			})?.text();
+
+			if (varName) {
+				for (const factory of ZLIB_FACTORIES) {
+					factoryBindings.push(`${varName}.${factory}`);
+				}
+			}
+		}
+	}
+
+	// If any import is found it's mean we can skip transformation on this file
+	if (!importNodes.length && dynamicImports.length === 0) return null;
+
+	// 2 Track variables assigned from factories (const, let, var)
+	for (const factory of factoryBindings) {
+		const patterns = [
+			`const $$$VAR = ${factory}($$$ARGS)`,
+			`let $$$VAR = ${factory}($$$ARGS)`,
+			`var $$$VAR = ${factory}($$$ARGS)`
+		];
```

Comment:
You need to use binding that have been found for each "factory" because it can have a different name, and resolve binding path, will have the named used in the file.

Variable name can be preserved, I think we don't want to change user variable name at the end.